elki-core-math/src/main/java/elki/math/statistics/dependence/mcde/MWPTest.java

```diff
@@ -178,14 +190,35 @@ public double statisticalTest(int start, int end, boolean[] slice, MWPRanking co
   }
 
   /**
-   * Parameterizer, returning the static instance.
-   * 
-   * @author Erich Schubert
+   * Adjusts idx so that it lies not in the middle of a tie by searching the ranks index
+   *
+   * @param idx   value to adjust for tie. In MCDE MWP the start and end of a slice
+   * @param ranks the corrected ranks index
+   * @return adjusted idx
    */
-  public static final class Par implements Parameterizer {
-    @Override
-    public MWPTest make() {
-      return STATIC;
+
+  protected int getSafeCut(int idx, MWPRanking ranks) {
+    double[] ref = ranks.adjusted;
+    final int len = ranks.index.length;
+    int i = 0;
+    while(true) {
+      if(idx - i - 1 < 0) {
```

Comment:
I'd prefer something like `int i = idx, j = idx;` and then move i backwards and j forwards?
This would make it easier to understand these boundary conditions. They likely could be combined:
```
for (int i=idx, j=idx, last = len - 1; ; i++, j++) {
if (i == 0 || ref[i] != ref[i - 1]) {
  return i;
}
if (j == last || ref[j] != ref[j + 1]) {
  return j;
}
}
```
also, I'd use len = ranks.adjusted.length; and maybe only pass the ranks.adjusted array to the function not the entire object; this makes it easier to see that this only depends on the adjusted ranks.